### PR TITLE
Add ocl-icd-system shim package

### PR DESCRIPTION
**Pull request by jaimergp**
Following conversations had in https://github.com/conda-forge/openmm-feedstock/pull/3#issuecomment-540083831 and https://github.com/openmm/openmm/issues/2401, this shim package will create the needed symlinks to make system OpenCL installations visible to `ocl-icd`. Hopefully this is a good compromise for everyone involved!

I am not sure about the name, so please feel free to suggest better options. Also, what license applies here?

__Checklist__

- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [X] Source is from official source
- [X] Package does not vend other packages
- [X] Build number is 0
- [ ] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

File: recipes/ocl-icd-system/activate.sh [hunk 1]
```diff
@@ -0,0 +1,12 @@
+conda_ocl_icd_system_activate () {
+    # If no OpenCL ICD has been installed with conda yet
+    if [[ -z "$(ls ${CONDA_PREFIX}/etc/OpenCL/vendors/)" ]]; then
+        # and a system installation is available
+        if [[ ! -z "$(ls /etc/OpenCL/vendors/*.icd 2> /dev/null || true)" ]]; then
+            ln -s /etc/OpenCL/vendors "${CONDA_PREFIX}"/etc/OpenCL/vendors/system
```
Comment (isuruf) [inline recipes/ocl-icd-system/activate.sh]: I would create this symlink irrespective of whether there's a system installation or not.

File: recipes/ocl-icd-system/activate.sh [hunk 2]
```diff
@@ -0,0 +1,9 @@
+conda_ocl_icd_system_activate () {
+    # If no OpenCL ICD has been installed with conda yet
+    if [[ -z "$(ls ${CONDA_PREFIX}/etc/OpenCL/vendors/)" ]]; then
+        ln -s /etc/OpenCL/vendors "${CONDA_PREFIX}"/etc/OpenCL/vendors/system
+        touch "${CONDA_PREFIX}"/etc/OpenCL/vendors/.ocl_icd_system_cf
```
Comment (isuruf) [inline recipes/ocl-icd-system/activate.sh]: Why this ?

File: recipes/ocl-icd-system/activate.sh [hunk 3]
```diff
@@ -0,0 +1,9 @@
+conda_ocl_icd_system_activate () {
+    # If no OpenCL ICD has been installed with conda yet
+    if [[ -z "$(ls ${CONDA_PREFIX}/etc/OpenCL/vendors/)" ]]; then
```
Comment (isuruf) [inline recipes/ocl-icd-system/activate.sh]: Same as before. No need to check this.

File: recipes/ocl-icd-system/activate.sh [hunk 4]
```diff
@@ -0,0 +1,9 @@
+conda_ocl_icd_system_activate () {
```
Comment (isuruf) [inline recipes/ocl-icd-system/activate.sh]: Can you make this a `post-link.sh` script and the deactivate a `pre-unlink.sh` script.

File: recipes/ocl-icd-system/activate.sh (hunk 2 above)
Comment (jaimergp) [reply recipes/ocl-icd-system/activate.sh]: To make sure it was created by this package, and not by the user. It would be very unlucky, but possible?

Comment (isuruf) [reply recipes/ocl-icd-system/activate.sh]: Why not do change the name from `system` to `system_cf` or something similar.

Comment (jaimergp) [reply recipes/ocl-icd-system/activate.sh]: What about `ocl-icd-system` so it is consistent with the package name?

Comment (isuruf) [reply recipes/ocl-icd-system/activate.sh]: Sure.

**Comment by jaimergp** [comment]
That last commit addresses all suggestions. If we all agree on the name and descriptions in `meta.yaml`, I am happy with the result. Also, do I need to include a license?

**Comment by isuruf** [comment]
You can add https://github.com/conda-forge/staged-recipes/blob/master/LICENSE.txt as a license.

**Comment by jakirkham** [comment]
cc @peastman

**Pull request merged**